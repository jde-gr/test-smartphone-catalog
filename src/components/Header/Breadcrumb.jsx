import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { BreadcrumbContainer } from './styles/Breadcrumb.styles';

const Breadcrumb = (props) => {
  const location = useLocation();

  return (
    <BreadcrumbContainer>
      <Link key={'/'} to={'/'}>
        <span>{'> '}Inicio</span>
      </Link>
      {!!location.state?.device && (
        <span>
          {' '}
          {' > '}
          {location.state.device}
        </span>
      )}
    </BreadcrumbContainer>
  );
};

Breadcrumb.propTypes = {};

export default Breadcrumb;
