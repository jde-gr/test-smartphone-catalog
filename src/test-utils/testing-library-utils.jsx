import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
