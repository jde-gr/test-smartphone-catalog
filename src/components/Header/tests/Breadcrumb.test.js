import { render, screen } from '../../../test-utils/testing-library-utils';
import Breadcrumb from '../Breadcrumb';

test('renders correctly', () => {
  const { container } = render(<Breadcrumb />);
  expect(container).toMatchSnapshot();
});

test('go back link links to the device list page', async () => {
  render(<Breadcrumb />);

  const homeLink = await screen.findByRole('link', { name: /inicio$/i });
  expect(homeLink).toHaveAttribute('href', '/');
});
