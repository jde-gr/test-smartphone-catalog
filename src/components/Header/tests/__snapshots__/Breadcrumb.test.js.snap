// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-y7aohw"
  >
    <a
      href="/"
    >
      <span>
        &gt; 
        Inicio
      </span>
    </a>
  </div>
</div>
`;
